tests/aws/scenario/kinesis_firehose/conftest.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+import pytest
+
+from localstack.utils.sync import retry
+
+
+@pytest.fixture
+def read_s3_data(aws_client):
+    s3 = aws_client.s3
+
+    def _read_s3_data(bucket_name: str, timeout: int = 10) -> dict[str, str]:
+        def _get_data():
+            response = s3.list_objects(Bucket=bucket_name)
+            if response.get("Contents") is None:
+                raise Exception("No data in bucket yet")
+
+            keys = [obj.get("Key") for obj in response.get("Contents")]
+
+            bucket_data = dict()
+            for key in keys:
+                response = s3.get_object(Bucket=bucket_name, Key=key)
+                data = response["Body"].read().decode("utf-8")
+                bucket_data[key] = data
+            return bucket_data
+
+        bucket_data = retry(_get_data, sleep=1, retries=timeout)
```

Comment:
What if new objects are created after the first successful retry iteration? Doesn't this mean that some objects will be missed?

Reply:
this simple get_data is only used if you want to fetch the first (or only) data from a bucket, if you want to fetch all or more data, you would need the get_all_messages